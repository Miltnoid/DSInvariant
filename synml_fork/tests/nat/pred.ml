type nat =
  | O
  | S of nat

let pred : nat -> nat |>
  { O => O
  ; S (O) => O
  ; S (S (O)) => S (O) } = ?
