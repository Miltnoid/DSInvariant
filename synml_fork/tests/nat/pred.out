(* Generated Mon Dec 29 00:37:32 2014 *)
let pred : nat -> nat =
  fun (n1:nat) -> match n1 with
                    | O -> O
                    | S n2 -> n2
;;
